tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/HotChocolateCommon.cs

```diff
@@ -0,0 +1,163 @@
+// <copyright file="HotChocolateCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate
+{
+    internal class HotChocolateCommon : GraphQLCommonBase
+    {
+        internal const string IntegrationName = nameof(Configuration.IntegrationId.HotChocolate);
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.HotChocolate;
+
+        private const string ServiceName = "hotchocolate";
+        internal const string ErrorType = "HotChocolate.Error";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(HotChocolateCommon));
+
+        internal static Scope CreateScopeFromExecuteAsync(Tracer tracer, IQueryRequest request)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+            try
+            {
+                var queryOperationName = request.OperationName;
+                var source = request.Query?.ToString();
+                var operationType = "Uncompleted";
+                scope = CreateScopeFromExecuteAsync(tracer, IntegrationId, new HotChocolateTags(), ServiceName, queryOperationName, source, operationType);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error creating or populating scope.");
+            }
+
+            return scope;
+        }
+
+        internal static Scope UpdateScopeFromExecuteAsync(Tracer tracer, IOperationContext context)
+        {
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            var scope = tracer.InternalActiveScope;
+            var span = scope?.Span;
+            if (span == null || span.OperationName != ExecuteOperationName)
+            {
+                // not in a Hotchocolate execution span
+                return null;
+            }
+
+            try
+            {
+                var operation = context.Operation;
+                var operationType = operation.OperationType.ToString();
+                var operationName = span.GetTag(Trace.Tags.GraphQLOperationName);
+                span.ResourceName = $"{operationType} {operationName ?? "operation"}";
+                span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
```

Comment:
nit: `GetTag()` and `SetTag()` are slow compared to accessing the properties directly - given we know this is a hot chocolate span I think it's safe enough to cast here and access the tags directly? Can add fallback though if there are edge cases I'm missing?
```suggestion
                if(span.Tags is HotChocolateTags tags)
                {
                    span.ResourceName = $"{operationType} {tags.GraphQLOperationName ?? "operation"}";
                    tags.GraphQLOperationType = operationType;
                }
                else
                {
                    var operationName = span.GetTag(Trace.Tags.GraphQLOperationName);
                    span.ResourceName = $"{operationType} {operationName ?? "operation"}";
                    span.SetTag(Trace.Tags.GraphQLOperationType, operationType);
                }
                

                
```